app/src/main/java/com/hedvig/app/feature/onbarding/ui/MoreOptionsAdapter.kt

```diff
@@ -0,0 +1,137 @@
+package com.hedvig.app.feature.onbarding.ui
+
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.hedvig.app.BuildConfig
+import com.hedvig.app.R
+import com.hedvig.app.databinding.MoreOptionsRowBinding
+import com.hedvig.app.databinding.SettingsBinding
+import com.hedvig.app.feature.onbarding.MoreOptionsModel
+import com.hedvig.app.feature.onbarding.MoreOptionsViewModel
+import com.hedvig.app.feature.settings.SettingsActivity
+import com.hedvig.app.util.GenericDiffUtilItemCallback
+import com.hedvig.app.util.extensions.dp
+import com.hedvig.app.util.extensions.inflate
+import com.hedvig.app.util.extensions.invalid
+import com.hedvig.app.util.extensions.setCompoundDrawable
+import com.hedvig.app.util.extensions.view.setHapticClickListener
+import com.hedvig.app.util.extensions.viewBinding
+
+class MoreOptionsAdapter(private val viewModel: MoreOptionsViewModel) :
+    ListAdapter<MoreOptionsModel, MoreOptionsAdapter.ViewHolder>(GenericDiffUtilItemCallback()) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = when (viewType) {
+        HEADER -> ViewHolder.Header(parent)
+        MEMBER_ID_SUCCESS -> ViewHolder.UserIdSuccess(parent)
+        MEMBER_ID_ERROR -> ViewHolder.UserIdError(parent)
+        VERSION -> ViewHolder.Version(parent)
+        SETTINGS -> ViewHolder.Settings(parent)
+        COPYRIGHT -> ViewHolder.Copyright(parent)
+        else -> throw Error("Unreachable")
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bind(getItem(position), viewModel)
+    }
+
+    override fun getItemViewType(position: Int) = when (getItem(position)) {
+        MoreOptionsModel.Header -> HEADER
+        is MoreOptionsModel.UserId.Success -> MEMBER_ID_SUCCESS
+        MoreOptionsModel.UserId.Error -> MEMBER_ID_ERROR
+        MoreOptionsModel.Version -> VERSION
+        MoreOptionsModel.Settings -> SETTINGS
+        MoreOptionsModel.Copyright -> COPYRIGHT
```

Comment:
Another approach that can be used here is to have `MoreOptionsModel.Row.{UserId|Version|Settings}`, given that they all use the same layout. You can then match on the model type in a common `ViewHolder`, and bind them accordingly there. That's what I've usually done in these cases 😉